app/Http/Controllers/AcceptancesController.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Acceptance;
+use App\Submission;
+use Illuminate\Http\Request;
+
+class AcceptancesController extends Controller
+{
+    public function store(Request $request)
+    {
+        $submission = Submission::findOrFail($request->input('submissionId'));
```

Comment:
We should guard to make sure this user has permissions on this submission.

Reply:
Maybe just auth()->user()->submissions()->findOrFail()

Reply:
That relationship is... involved. 

`user -> talks[] -> revisions[] -> submission -> acceptance->findOrFail().`

I tried a different way of guarding instead. Very open to your opinion there as I'm sure there's a nicer way to do it.

Reply:
Geez, so that's a user has many through many through, which definitely doesn't exist natively.

I don't see an existing guard in here--is there one? I don't see anything that ensures that the currently-logged-in user has access to the submission.

Let's try it from the other angle: check that the `$submission` is owned by current user.

```php
$submission = Submission::findOrFail($request->input('submissionId'));

if (auth()->user()->id !== $submssion->talkRevision->talk->user_id) {
    abort(401);
}
```

maybe?

Reply:
```
 $userHasAccess = auth()->user()->talks->filter(function ($talk) use ($submission) {
            return $talk->talkRevision === $submission->talkRevision;
        }) !== false;
        if (!$userHasAccess) {
            throw new Exception("Not Authorized");
        }
```

That's what I did, but I think your suggestion is probably a better one.

Reply:
Yah.. I didn't see yours because of the outdated diff. Sigh. Let me take a look.

Reply:
Yah I think mine's a bit clearer but also makes less database calls so let's go with that --thanks!